### PR TITLE
Render code blocks in repo description

### PR DESCRIPTION
**Pull request by silverwind**
Backtick syntax now works in repo description too. Also, I replaced the CSS for this was a new single class, making it more flexible and not dependent on a parent. Also, very slightly reduced font size from 16.8px to 16px.

Screenshot 2023-08-31 at 00 47 52

### CODE REVIEW

File: templates/repo/home.tmpl [hunk 1]
```diff
@@ -8,7 +8,7 @@
 		<div class="ui repo-description">
 			<div id="repo-desc">
 				{{$description := .Repository.DescriptionHTML $.Context}}
-				{{if $description}}<span class="description">{{$description}}</span>{{else if .IsRepositoryAdmin}}<span class="no-description text-italic">{{.locale.Tr "repo.no_desc"}}</span>{{end}}
+				{{if $description}}<span class="description">{{$description | RenderCodeBlock}}</span>{{else if .IsRepositoryAdmin}}<span class="no-description text-italic">{{.locale.Tr "repo.no_desc"}}</span>{{end}}
```
Comment (delvh) [inline templates/repo/home.tmpl:11]: Surprising that that passes the validation…
I would expect that to escape the HTML tags as it is (obviously) unsafe…

**Review by delvh — APPROVED** [review]

File: templates/repo/home.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/repo/home.tmpl:11]: You are right, it's possible to inject HTML here, wonder if that was the case before. Any suggestions how to fix?

Comment (silverwind) [reply templates/repo/home.tmpl:11]: Can confirm HTML injection was possible before. It's sanitized but imho we should remove this.

Comment (silverwind) [reply templates/repo/home.tmpl:11]: Could also be that this injection is somewhat intentional if the title is rendered like other markdown. Question is really if we want this thought as it allows the user to break page layout in the repo title, which imho is to be avoided.

File: web_src/css/repo.css [hunk 2]
```diff
@@ -236,8 +236,8 @@
   }
 }
 
-.repository.file.list #repo-desc {
-  font-size: 1.2em;
+#repo-desc {
+  font-size: 16px;
```
Comment (wxiaoguang) [inline web_src/css/repo.css]: IIRC, you have added `gt-font-16` in #24812 , could the `gt-font-16` be used to replace this style?

Comment (silverwind) [reply web_src/css/repo.css]: done

**Comment by silverwind** [comment]
I guess we can merge this now and defer the discussion about HTML rendering in repo description to later, to not expand the scope of this change.

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**